File: Solo/Codesizes/Easy/don't-panic/DontPanic.rb
STDOUT.sync=true
_,_,_,eF,eP,_,_,E=gets.split.map(&:to_i)
D=Hash[E.times.map{gets.split.map(&:to_i)}]
D[eF]=eP
loop{cF,cP,di=gets.split;puts cF.to_i<0?"WAIT":((di[0]=="L"&&D[cF.to_i]>cP.to_i)||(di[0]=="R"&&D[cF.to_i]<cP.to_i))?"BLOCK":"WAIT"}

File: Solo/Codesizes/Easy/don't-panic/DontPanic.py
nF,W,R,eF,eP,tC,aE,E=map(int,input().split())
D={}
for _ in' '*E:f,p=map(int,input().split());D[f]=p
D[eF]=eP
while 1:
 cF,cP,di=input().split();cF=int(cF);cP=int(cP)
 print("WAIT"if cF<0 else("BLOCK"if(di[0]=="L"and D[cF]>cP or di[0]=="R"and D[cF]<cP)else"WAIT"))
